pkg/driver/driver.go

```diff
@@ -645,4 +688,96 @@ func (driver *Driver) IsNFSResourceRequest(parameters map[string]string) bool {
 	return false
 }
 
+// ScrubberTask executes the scrubber function
+func (driver *Driver) ScrubberTask(t time.Time) error {
+	log.Infof("Scrubber task invoked at %v", t)
+	if err := driver.ScrubEphemeralPods(); err != nil {
+		log.Error(err.Error())
+		// Log error and continue
+	}
+	log.Infof("Scrubber task completed at %v", time.Now())
+	return nil
+}
+
+// ScrubEphemeralPods to cleanup the orphan/stale ephemeral volumes and its associated staged devices
+func (driver *Driver) ScrubEphemeralPods() error {
+	log.Trace(">>>>> ScrubEphemeralPods")
+	defer log.Trace("<<<<< ScrubEphemeralPods")
+
+	// Check if pods path exists
+	exists, isDir, err := util.FileExists(podsDirPath)
+	if err != nil {
+		return err
+	}
+	if !exists || !isDir {
+		return fmt.Errorf("Path [%s] does not exist or not a directory", podsDirPath)
+	}
+
+	// Fetch all ephemeral pods and populate it
+	//ephemeralPods := map[string][]*StagingDeviceEphemeralData{}
```

Comment:
Can this line be removed?

Reply:
do we expect more than 1 VolumeHandleTargetPath for a single podID? I had thought we have 1-1 mapping between pod and volume. In case of failover too a new pod is created for which we create a new volume.

Reply:
If single pod has two volumes, then we will have two target paths, one for each volume.
ex:
/var/lib/kubelet/pods/b627d92c-022b-461f-84a3-2163b033bd6f/volumes/kubernetes.io~csi/my-volume-vols-1/mount
/var/lib/kubelet/pods/b627d92c-022b-461f-84a3-2163b033bd6f/volumes/kubernetes.io~csi/my-volume-vols-2/mount